Fix data augment

Alcharyx opened the pull request.

Alcharyx committed:
fix: augment_all_vid() & grid_augment()

augment_all_vid() remove files named "readme.md" from list of folder to look at instead of go until len(folder)-2
grid_augment() add video name variable

Alcharyx committed:
fix: augment_all_vid()

better usage of path class

Alcharyx committed:
feat: add tqdm library to project

Alcharyx committed:
refactor: move augment_all_vid to dataset.py

use _fps_folder_name()
add random to dataset.py
add json_name parameter in augment_all_vid()

Alcharyx committed:
refactor: move list link into BK class

Change plot skeleton to use BK class

Alcharyx committed:
fix: correct concatenation of Path class

MarcBresson commented on tactus_data/utils/data_augment.py:
```diff
@@ -113,7 +112,7 @@ def flip_h_2d(input_folder_path: Path,
     """
     for file_name in json_name:
-        with open(str(input_folder_path) + "\\" + file_name) as file:
+        with open(str(input_folder_path / file_name)) as file:
```
I  think open() accepts Path objects, but you could also do (input_folder_path / file_name).open()

MarcBresson commented on tactus_data/utils/data_augment.py:
```diff
@@ -457,11 +457,12 @@ def grid_augment(path_json: Path,
         original_data = json.load(file)
     for indices in generator:
         if flag_limit and generated_pic >= max_copy:
+            print("Max copy overflow")
```
is it something that happens a lot ?

MarcBresson commented:
thank you for the line breaks in datasets.py :D

MarcBresson committed:
Merge branch 'dev' into fix-data-augment

MarcBresson commented:
maybe you could create default grid presets that would serve as a default argument. For now, the only thing that tactus_data export is the ut_interaction module. I think it would be cool to just be able to call ut_interaction.augment_data() that calls the augment_all_vid() function from datasets.py

Alcharyx committed:
feat: (WIP) multiaugment()

multiaugment allow you to do all the previous augment in a cleaner way and 1 go
_cart_augment() do the augment inside cartesian dimension
_get_transform_matrix() returns the correct augment matrix depending on input parameters for cartesian augmentation

Alcharyx committed:
feat: gridParam class

Class used to create the grid for grid_augment()
init_test allows you to ini gridparam with the grid used for test conditions

Alcharyx committed:
refactor: adapt grid augment for multiaugment

Alcharyx committed:
refactor: erase old non used augment

Alcharyx committed:
doc: change doc of functions accordingly

remove the use of first value for grid augment generator

Alcharyx committed:
refactor: gridParam turn default value into non mutable argument tuple
